LayoutKit/Layouts/TextViewLayout.swift

```diff
@@ -0,0 +1,138 @@
+    }
+
+    // MARK: - Convenience initializers
+
+    public convenience init(text: String,
+                            font: UIFont = defaultFont,
+                            lineFragmentPadding: CGFloat = 0,
+                            textContainerInset: UIEdgeInsets = .zero,
+                            layoutAlignment: Alignment = defaultAlignment,
+                            flexibility: Flexibility = defaultFlexibility,
+                            viewReuseId: String? = nil,
+                            configure: ((TextView) -> Void)? = nil) {
+        self.init(text: .unattributed(text),
+                  font: font,
+                  lineFragmentPadding: lineFragmentPadding,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    public convenience init(attributedText: NSAttributedString,
+                            font: UIFont = defaultFont,
+                            lineFragmentPadding: CGFloat = 0,
+                            textContainerInset: UIEdgeInsets = .zero,
+                            layoutAlignment: Alignment = defaultAlignment,
+                            flexibility: Flexibility = defaultFlexibility,
+                            viewReuseId: String? = nil,
+                            configure: ((TextView) -> Void)? = nil) {
+        self.init(text: .attributed(attributedText),
+                  font: font,
+                  lineFragmentPadding: lineFragmentPadding,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    // MARK: - Layout protocol
+
+    open func measurement(within maxSize: CGSize) -> LayoutMeasurement {
+        let fittedSize = textSize(within: maxSize)
+        let decreasedToSize = fittedSize.decreasedToSize(maxSize)
+        return LayoutMeasurement(layout: self, size:decreasedToSize, maxSize: maxSize, sublayouts: [])
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    // MARK: - private helpers
+
+    private func textSize(within maxSize: CGSize) -> CGSize {
+        var size = text.textSize(maxSize: maxSize, font: font)
+        let heightInset = textContainerInset.top + textContainerInset.bottom
+        let widthInset = textContainerInset.left + textContainerInset.right + lineFragmentPadding * 2
+        size = CGSize(width: size.width + widthInset, height: size.height + heightInset)
```

Comment:
Should probably reduce the available size `maxSize` by the height and width insets instead of adding them to the resulting size.